presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperatorFactory.java

```diff
@@ -67,29 +68,30 @@ public LookupJoinOperatorFactory(int operatorId,
         this.joinType = requireNonNull(joinType, "joinType is null");
         this.joinProbeFactory = requireNonNull(joinProbeFactory, "joinProbeFactory is null");
 
-        this.referenceCount = new ReferenceCount();
+        probeReferenceCount = new ReferenceCount();
```

Comment:
This would be a big change. Moreover the transitions are practically arbitrary, as we need to care about probe being empty, build being empty and such. And we don't know how many outer operators there will be unless we count them. `ReferenceCount` conveys the purpose. I agree this should be refactored, but it would be unwise to undertake a refactor while other PRs are in flight in this area.